src/js/utils/styles.js

```diff
@@ -294,9 +303,11 @@ export const inputStyle = css`
           props.theme,
         )}
   ${props =>
-    props.theme.global.input.weight &&
+    // for backwards compatibility, check if props.theme.global.input.weight
+    (props.theme.global.input.weight || props.theme.global.input.font.weight) &&
```

Comment:
Communicating here our conversation resolutions.
In case the user defines `theme.global.input.weight`, applying  `props.theme.global.input.font.weight` won't work. we will add to documentation and communicate effectively that those two shouldn't be used together.